homeassistant/components/hive/water_heater.py

```diff
@@ -0,0 +1,112 @@
+"""Support for hive water heaters."""
+from homeassistant.const import TEMP_CELSIUS
+
+from homeassistant.components.water_heater import (
+    STATE_ECO, STATE_ON, STATE_OFF, SUPPORT_OPERATION_MODE, WaterHeaterDevice)
+
+from . import DATA_HIVE, DOMAIN
+
+SUPPORT_FLAGS_HEATER = (SUPPORT_OPERATION_MODE)
+
+HIVE_TO_HASS_STATE = {
+    'SCHEDULE': STATE_ECO,
+    'ON': STATE_ON,
+    'OFF': STATE_OFF,
+}
+
+HASS_TO_HIVE_STATE = {
+    STATE_ECO: 'SCHEDULE',
+    STATE_ON: 'ON',
+    STATE_OFF: 'OFF',
+}
+
+SUPPORT_WATER_HEATER = [STATE_ECO, STATE_ON, STATE_OFF]
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Wink water heater devices."""
+    if discovery_info is None:
+        return
+    if discovery_info["HA_DeviceType"] != "HotWater":
+        return
+
+    session = hass.data.get(DATA_HIVE)
+    water_heater = HiveWaterHeater(session, discovery_info)
+
+    add_entities([water_heater])
+    session.entities.append(water_heater)
+
+
+class HiveWaterHeater(WaterHeaterDevice):
+    """Hive Water Heater Device."""
+
+    def __init__(self, hivesession, hivedevice):
+        """Initialize the Water Heater device."""
+        self.node_id = hivedevice["Hive_NodeID"]
+        self.node_name = hivedevice["Hive_NodeName"]
+        self.device_type = hivedevice["HA_DeviceType"]
+        self.session = hivesession
+        self.data_updatesource = '{}.{}'.format(
+            self.device_type, self.node_id)
+        self._unique_id = '{}-{}'.format(self.node_id, self.device_type)
+        self._unit_of_measurement = TEMP_CELSIUS
+        self.session.entities.append(self)
```

Comment:
Why do we need to appen the entity again here? We do it in setup_platform already, right?

Reply:
We should probably move the append to `async_added_to_hass` to avoid calling `self.schedule_update_ha_state` before the entity has been added to home assistant.

Reply:
Hi @MartinHjelmare, since this PR has been merged I will remove the duplication in the next release.

Reply:
If you can make a PR asap, we can adjust this before 0.96 goes out.